samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -88,95 +98,13 @@ public class ChatSkill
-            Utilities.TokenCount(string.Join("\n", new string[]
-                {
-                    this._promptOptions.SystemDescription,
-                    this._promptOptions.SystemIntent,
-                    this._promptOptions.SystemIntentContinuation
-                })
-            );
-
-        // Clone the context to avoid modifying the original context variables.
-        var intentExtractionContext = Utilities.CopyContextWithVariablesClone(context);
-        intentExtractionContext.Variables.Set("tokenLimit", historyTokenBudget.ToString(new NumberFormatInfo()));
-        intentExtractionContext.Variables.Set("knowledgeCutoff", this._promptOptions.KnowledgeCutoffDate);
-
-        var completionFunction = this._kernel.CreateSemanticFunction(
-            this._promptOptions.SystemIntentExtraction,
-            skillName: nameof(ChatSkill),
-            description: "Complete the prompt.");
-
-        var result = await completionFunction.InvokeAsync(
-            intentExtractionContext,
-            settings: this.CreateIntentCompletionSettings()
-        );
-
-        if (result.ErrorOccurred)
-        {
-            context.Log.LogError("{0}: {1}", result.LastErrorDescription, result.LastException);
-            context.Fail(result.LastErrorDescription);
-            return string.Empty;
-        }
 
-        return $"User intent: {result}";
-    }
-
-    /// <summary>
-    /// Extract the list of participants from the conversation history.
-    /// Note that only those who have spoken will be included.
-    /// </summary>
-    [SKFunction, Description("Extract audience list")]
-    [SKParameter("chatId", "Chat ID to extract history from")]
-    public async Task<string> ExtractAudienceAsync(SKContext context)
-    {
-        var tokenLimit = this._promptOptions.CompletionTokenLimit;
-        var historyTokenBudget =
-            tokenLimit -
-            this._promptOptions.ResponseTokenLimit -
-            Utilities.TokenCount(string.Join("\n", new string[]
-                {
-                    this._promptOptions.SystemAudience,
-                    this._promptOptions.SystemAudienceContinuation,
-                })
-            );
-
-        // Clone the context to avoid modifying the original context variables.
-        var audienceExtractionContext = Utilities.CopyContextWithVariablesClone(context);
-        audienceExtractionContext.Variables.Set("tokenLimit", historyTokenBudget.ToString(new NumberFormatInfo()));
-
-        var completionFunction = this._kernel.CreateSemanticFunction(
-            this._promptOptions.SystemAudienceExtraction,
-            skillName: nameof(ChatSkill),
-            description: "Complete the prompt.");
-
-        var result = await completionFunction.InvokeAsync(
-            audienceExtractionContext,
-            settings: this.CreateIntentCompletionSettings()
-        );
+        var parentDir = System.IO.Directory.GetCurrentDirectory() + "\\..\\..\\..";
```

Comment:
nit: abstract this out to have a safer implementation. Refer to an existing implementation here: https://github.com/microsoft/semantic-kernel/blob/main/samples/dotnet/KernelHttpServer/Utils/RepoFiles.cs